transaction-view/src/sanitize.rs

```diff
@@ -0,0 +1,525 @@
+use crate::{
+    result::{Result, TransactionViewError},
+    transaction_data::TransactionData,
+    transaction_view::UnsanitizedTransactionView,
+};
+
+pub(crate) fn sanitize(view: &UnsanitizedTransactionView<impl TransactionData>) -> Result<()> {
```

Comment:
actual sanitization checks done here. Data movement for type-change done in `TransactionView`.